models/_sector/dex/trades/optimism/platforms/saddle_finance_optimism_base_trades.sql

```diff
@@ -0,0 +1,60 @@
+{{ config(
+    schema = 'saddle_finance_optimism'
+    ,alias = 'base_trades'
+    ,materialized = 'incremental'
+    ,file_format = 'delta'
+    ,incremental_strategy = 'merge'
+    ,unique_key = ['tx_hash', 'evt_index']
+    ,incremental_predicates = [incremental_predicate('DBT_INTERNAL_DEST.block_time')]
+    )
+}}
+
+{% set saddle_finance_optimism_evt_trade_tables = [
+    source('saddle_finance_optimism', 'MetaSwap_evt_TokenSwap')
+] %}
+
+with dexs AS (
+    {% for evt_trade_table in saddle_finance_optimism_evt_trade_tables %}
+        SELECT
+            evt_block_time          AS block_time,
+            evt_tx_to               AS taker,
+            evt_tx_from             AS maker,
+            tokensBought            AS token_bought_amount_raw,
+            tokensSold               AS token_sold_amount_raw,
+            CAST(NULL AS double)    AS amount_usd,
+            CAST(NULL AS varbinary)                 AS token_bought_address,
+            CAST(NULL AS varbinary)                AS token_sold_address,
```

Comment:
These two fields should be not null if I am not wrong.
Do you have any specific reason to make them null?

Reply:
Yes. In the decode table of saddle_finance, we do not have these two columns that's why I have added these two columns as null.

Reply:
In this case we may need to find other sources to join, maybe log?

Reply:
Yes, it seems better to retrieve those two columns from the log source table. If possible, could you provide any reference or guidance on how to accomplish this?

Reply:
That might be a little bit complicated, but this should be an example:
https://github.com/duneanalytics/spellbook/blob/main/models/hashflow/ethereum/hashflow_ethereum_raw_trades.sql

Anyway if we can submit new contracts for decoding and use them as decoded sources, it will be easier.

Reply:
You may need to fill in these columns first, as they are not-null columns by requirements.

Reply:
Hello @Hosuke ,

As of now, can I remove "saddle_finance" from this PR? Once it is fully ready, I will submit another PR for it.

Reply:
Yes, you can remove `saddle_finance` first.

Reply:
Done.